### DISCUSSION TITLE
Using `forms.NewSettingsUpsert` in golang migration

### DISCUSSION THREAD

**Discussion by abiriadev** [pocketbase/pocketbase#4625] [Q&A] [answered]
I would like to configure my application with migration, so I chose to use `forms.NewSettingsUpsert` here. The problem is, I can't access `core.App` in any way. Is there a way to do this?

```go
package migrations

import (
	"github.com/pocketbase/dbx"
	"github.com/pocketbase/pocketbase/forms"
	m "github.com/pocketbase/pocketbase/migrations"
)

func init() {
	m.Register(func(db dbx.Builder) error {
		forms.NewSettingsUpsert()

		return nil
	}, func(db dbx.Builder) error {
		// add down queries...

		return nil
	})
}
```

**Comment by ganigeorgiev** [comment, marked as answer]
No, it is not possible at the moment (_or at least not without defining your `app` instance globally and even then you'll have to change the form dao to the transaction one_).
This will be supported with https://github.com/pocketbase/pocketbase/discussions/4355.

**Reply by abiriadev** [reply]
Thanks! Is it ok to use `dao.SaveSettings` instead in this case?

**Reply by ganigeorgiev** [reply]
Yes, it is OK (you can also check the [example from the docs](https://pocketbase.io/docs/go-migrations/#initialize-default-application-settings)).
